crates/aiken-project/src/lib.rs

```diff
@@ -702,14 +706,51 @@ where
             cpu: i64::MAX,
         };
 
-        scripts
-            .into_iter()
-            .filter(|script| -> bool {
-                let path = format!("{}{}", script.module, script.name);
+        let scripts = if let Some(match_tests) = match_tests {
+            let match_tests: Vec<(&str, Option<Vec<String>>)> = match_tests
+                .iter()
+                .map(|match_test| {
+                    let mut match_split_dot = match_test.split('.');
 
-                !matches!(&match_name, Some(search_str) if !path.contains(search_str))
-            })
-            .collect::<Vec<Script>>()
+                    let match_module = match_split_dot.next().unwrap_or("");
+
+                    let match_names = match_split_dot.next().map(|names| {
+                        let names = names.replace(&['{', '}'][..], "");
+
+                        let names_split_comma = names.split(',');
+
+                        names_split_comma.map(str::to_string).collect()
+                    });
+
+                    (match_module, match_names)
+                })
+                .collect();
+
+            scripts
+                .into_iter()
+                .filter(|script| -> bool {
+                    match_tests.iter().any(|(match_module, match_names)| {
+                        if *match_module == script.module {
+                            if let Some(match_names) = match_names {
+                                match_names
+                                    .iter()
+                                    .any(|name| name == &script.name || script.name.contains(name))
```

Comment:
Isn't the left-side of `||` redundant with the right side?

Reply:
Not necessarily.

For example 

"flatten" == "flatten_2"

vs.

"flatten_2".contains("flatten")

The initial == is so that you can direct match a single test in a module. The contains is to run whatever is kinda similar.

Reply:
I mean, if the left-hand side is true, then necessarily the right-hand side is also true; in this very case:

`name == foo` _implies_ `foo.contains(name)` 


Given that they are in disjunction `||`, then the left-hand side is effectively redundant and it suffices to have only the right-hand side, isn't it 😅 ?

Reply:
Oh... yea nvm :)